vital/util/buffer.h

```diff
@@ -0,0 +1,56 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief Base interface for simple buffer classes
+ */
+
+#ifndef KWIVER_VITAL_UTIL_BUFFER_H_
+#define KWIVER_VITAL_UTIL_BUFFER_H_
+
+namespace kwiver {
+namespace vital {
+
+
+/// Base interface for simple buffer classe
+template< class Data >
```

Comment:
I'm curious why we need an abstract base class for buffers of a fixed type?  What are the planed derived classes other than a ring buffer?  Why do we need polymorphism here?  It would seem better to just make an STL style ring buffer class.